src/coreclr/jit/lower.cpp

```diff
@@ -7901,6 +7901,11 @@ PhaseStatus Lowering::DoPhase()
         comp->lvSetMinOptsDoNotEnreg();
     }
 
+    if (comp->opts.OptimizationEnabled())
+    {
+        MapParameterRegisterLocals();
+    }
+
```

Comment:
There's a problem with changing IR after lowering has run because some nodes may have made containment decisions based on seeing a `LCL_FLD` of a particular size. I think the problem I was seeing was for compares on xarch.
I moved it to happen before lowering for now, but this may need to be moved back to happen after in the future (I think to fix #112138 we'll need that).